app/controllers/api/automated_deploys_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+# deploys that automatically get triggered when new hosts come up or are restarted
+class Api::AutomatedDeploysController < Api::BaseController
+  STAGE_NAME = 'Automated Deploys'
+
+  before_action :find_or_create_stage
+  before_action :find_deploy_group
+  before_action :find_last_deploy
+
+  def create
+    deploy_service = DeployService.new(current_user)
+    deploy = deploy_service.deploy!(
+      @stage,
+      reference: @last_deploy.reference,
+      buddy_id: @last_deploy.buddy_id,
```

Comment:
we use the same copy-the-buddy approach when deploying an already approved version to other stages, so I don't think we should just reuse that approach ... if we want to do something else that can be a separate PR and compliance discussion for what they'd want to see